requests_toolbelt/utils/dump.py

```diff
@@ -3,19 +3,116 @@
 
 from requests import compat
 
-
-__all__ = ('dump_response', 'dump_all')
+__all__ = ('dump_response', 'dump_all', 'Sanitizer', 'HeaderSanitizer')
 
 HTTP_VERSIONS = {
     9: b'0.9',
     10: b'1.0',
     11: b'1.1',
 }
 
+#: List of sensitive headers copied from
+#: https://github.com/google/har-sanitizer
+SENSITIVE_HEADERS = {
+    "state",
+    "shdf",
+    "usg",
+    "password",
+    "email",
+    "code",
+    "code_verifier",
+    "client_secret",
+    "client_id",
+    "token",
+    "access_token",
+    "authenticity_token",
+    "id_token",
+    "appid",
+    "challenge",
+    "facetid",
+    "assertion",
+    "fcparams",
+    "serverdata",
+    "authorization",
+    "auth",
+    "x-client-data",
+    "samlrequest",
+    "samlresponse"
+}
+
 _PrefixSettings = collections.namedtuple('PrefixSettings',
                                          ['request', 'response'])
 
 
+class Sanitizer(object):
+    """Performs no sanitation"""
+
+    def request_header(self, name, value):
+        """Sanitize a request header
+
+        :param name: The header name
+        :type name: `compat.basestring`
```

Comment:
It references the type `compat.basestring`, which is already in use in this module. However Sphinx doesn't know it, and since I'm using `bytes` elsewhere I guess the convention is to use Python 3 types. I'll change it to `str`.

Reply:
Strictly speaking I'm not 100% sure what type is expected here. It depends on what types are used in the headers. I believe it is `unicode` in Python 2 and `str` in Python 3, but I'm not sure.

Reply:
Generally speaking, it's roughly `Union[six.bytes_type, six.text_type]` where on Python 2 those are `str` and `unicode` respectively and on Python 3 they are `bytes` and `str`.

Reply:
So `requests.compat.basestring` is probably the most correct thing to use then?